### PR TITLE
fleshing out what’s on service

### PR DESCRIPTION
**Pull request by gestchild**
Splitting #1993 into more manageable chunks (will ultimately need all merged into master together, in order not to break anything. Once each PR is individual approved I'll merge them back down the chain)

### CODE REVIEW

File: whats-on/app/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM node:8-alpine
+COPY app.js app.js
+
+EXPOSE 3000
+
+CMD ["node", "app"]
```
Comment (jamesgorrie) [inline whats-on/app/Dockerfile]: Not sure this is needed - I think we use the one from the `./build` dir

File: whats-on/nginx/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,2 @@
+FROM nginx:alpine
+COPY nginx.conf /etc/nginx/nginx.conf
```
Comment (jamesgorrie) [inline whats-on/nginx/Dockerfile]: Not sure we need the `nginx` folder all together as we use the `./build` dir.

File: whats-on/terraform/services.tf [hunk 3]
```diff
@@ -0,0 +1,34 @@
+module "whats_on" {
+  source             = "git::https://github.com/wellcometrust/terraform.git//services?ref=v1.0.4"
```
Comment (jamesgorrie) [inline whats-on/terraform/services.tf]: If we use the `eventbrite` service as an example, we use the new version of the modules (v3)

File: whats-on/terraform/terraform.tf [hunk 4]
```diff
@@ -0,0 +1,41 @@
+terraform {
+  required_version = ">= 0.9"
```
Comment (jamesgorrie) [inline whats-on/terraform/terraform.tf]: See above about versioning.

File: whats-on/terraform/iam.tf [hunk 5]
```diff
@@ -0,0 +1,4 @@
+module "ecs_events_iam" {
```
Comment (jamesgorrie) [inline whats-on/terraform/iam.tf]: There's a few naming things here re `s/events/whats_on`.

File: whats-on/terraform/services.tf [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  name               = "whats-on"
```
Comment (jamesgorrie) [inline whats-on/terraform/services.tf]: Any chance we could use `_`?
It just makes it way more consistent with any of the other infra stuff, where dashes can be annoying.

Maybe in the folder name too?

File: whats-on/terraform/services.tf (hunk 3 above)
Comment (gestchild) [reply whats-on/terraform/services.tf]: As discussed, will keep this consistent with the event and exhibitions services and then we can look at updating them all together

File: whats-on/terraform/services.tf [hunk 7]
```diff
@@ -0,0 +1,34 @@
+module "whats_on" {
+  source             = "git::https://github.com/wellcometrust/terraform.git//services?ref=v1.0.4"
+  name               = "whats_on"
+  cluster_id         = "${local.cluster_name}"
+  task_role_arn      = "${module.ecs_whats_on_iam.task_role_arn}"
+  template_name      = "default"
+  vpc_id             = "${local.vpc_id}"
+  nginx_uri          = "wellcome/wellcomecollection_whats_on_nginx:${var.container_tag}"
+  app_uri            = "wellcome/wellcomecollection_whats_on_webapp:${var.container_tag}"
```
Comment (jamesgorrie) [inline whats-on/terraform/services.tf]: You'll need to be sure to create these in docker hub, and also allow write access to Circle.

File: whats-on/terraform/services.tf [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+  server_error_alarm_topic_arn = "${module.alb_server_error_alarm.arn}"
+  client_error_alarm_topic_arn = "${module.alb_client_error_alarm.arn}"
+
+  # These account for the 128 mem and CPU the nginx container use
+  # 995 is how much memmory is left once docker is running
+  cpu                      = "384" # (1024/2) - 128
+  memory                   = "369" # (995/2) - 128
+  primary_container_port   = "80"
+  secondary_container_port = "3002"
```
Comment (jamesgorrie) [inline whats-on/terraform/services.tf]: Have we updated the `app_cluster` to sustain this?

File: whats-on/terraform/terraform.tf [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+  backend "s3" {
+    key            = "build-state/whats-on.tfstate"
```
Comment (jamesgorrie) [inline whats-on/terraform/terraform.tf]: `_`?

**Pull request merged**